src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs

```diff
@@ -469,6 +464,10 @@ private async ValueTask<IInstalledAndTransitivePackages> GetInstalledAndTransiti
 
             if (itemFilter == ItemFilter.All)
             {
+                IInstalledAndTransitivePackages browseTabPackages = await GetInstalledAndTransitivePackagesAsync(projectContextInfos, useTransitiveOrigins: false, cancellationToken);
```

Comment:
Wouldn't this need to be useTransitiveOrigins: true, given that we are instantiating transitivePackageCollection?

Reply:
No, because this is the Browse tab and Browse tab doesn't use transitive origin data. I will add a comment

Reply:
This is unrelated to your changes, but the more I look at this code, the more I think we should be using design patterns to avoid all of the conditions in this implementation.